### DISCUSSION TITLE
Make MaxWorkspaceIndexFileCount a configurable variable

### DISCUSSION THREAD

**Discussion by shivk012** [microsoft/pyright#4603] [Ideas]
Simple proposal to turn MaxWorkspaceIndexFileCount at https://github.com/microsoft/pyright/blob/main/packages/pyright-internal/src/analyzer/program.ts#L95 into a variable that can be modified by a configuration file.
This limit is stopping the search by symbol in workspace (cmd + T) from working correctly in larger mono-repos.

**Comment by erictraut** [comment]
MaxWorkspaceIndexFileCount is used for file indexing, which is functionality that is not enabled (or even invoked) in pyright. There are pieces of indexing in place for use by pylance, Microsoft's premium Python language server. You can ignore MaxWorkspaceIndexFileCount and the indexing functionality.

If you're using pylance and would like to recommend changes to the indexing functionality, please file a issue or post to the discussion forum in the [pylance-release](https://github.com/microsoft/pylance-release) project.

**Reply by shivk012** [reply]
Thanks for the reply @erictraut. As the config for the indexing is implemented within pyright though I thought it would make sense to expose so things like pylance can modify if needed?

**Reply by erictraut** [reply]
If and when the pylance team decide to expose this, they will make the necessary change in pyright.
